src/relay/ethindex_db.py

```diff
@@ -255,49 +257,34 @@ def get_trustline_events(
         contract_address: str,
         user_address: str,
         counterparty_address: str,
-        event_name: str = None,
+        event_types: Iterable[str] = None,
         from_block: int = 0,
         timeout: float = None,
     ):
-        if event_name is None:
-            query = EventsQuery(
-                """blockNumber>=%s
-                   AND address=%s
-                   AND ((args->>'{_from}'=%s AND args->>'{_to}'=%s) OR (args->>'{_from}'=%s AND args->>'{_to}'=%s))
-                """.format(
-                    _from=self.from_to_types[event_name][0],
-                    _to=self.from_to_types[event_name][1],
-                ),
-                (
-                    from_block,
-                    event_name,
-                    contract_address,
-                    user_address,
-                    counterparty_address,
-                    counterparty_address,
-                    user_address,
-                ),
-            )
-        else:
-            query = EventsQuery(
-                """blockNumber>=%s
-                   AND eventName=%s
-                   AND address=%s
-                   AND ((args->>'{_from}'=%s AND args->>'{_to}'=%s) OR (args->>'{_from}'=%s AND args->>'{_to}'=%s))
-                """.format(
-                    _from=self.from_to_types[event_name][0],
-                    _to=self.from_to_types[event_name][1],
-                ),
-                (
-                    from_block,
-                    event_name,
-                    contract_address,
-                    user_address,
-                    counterparty_address,
-                    counterparty_address,
-                    user_address,
-                ),
-            )
+        event_types = self._get_standard_event_types(event_types)
+
+        all_from_to_types = set()
+        for from_, to in self.from_to_types.values():
+            all_from_to_types.add((from_, to))
+            all_from_to_types.add((to, from_))
+
+        from_to_where = " OR ".join(
+            f"(args->>'{from_}'=%s AND args->>'{to}'=%s)"
+            for from_, to in all_from_to_types
+        )
```

Comment:
I would use `a` and `b` or `user` and `counterparty` since they are not `from` and `to` anymore but could be `to` and `from`.

Also I don't like the name `from_to_where`, but maybe it's an SQL thing. Would prefer something like `from_to_request_filter` or `from_to_filter`

Reply:
WHERE is the sql keyword for filter, so thats why I use where